Lib/test/test_pkgutil.py

```diff
@@ -176,6 +176,21 @@ def test_walkpackages_zipfile(self):
                 continue
             del sys.modules[pkg]
 
+    def test_walkpackages_on_bad_input(self):
+        """Test that walk_packages() throws ValueError if path is str or bytes
+        object.
+        """
+        str_input = 'test_dir'
+        with self.assertRaises(ValueError) as context:
+            list(pkgutil.walk_packages(str_input))
+
+        self.assertTrue('path must be None or list of paths' in str(context.exception))
```

Comment:
I don't think this assertion is needed.  All we care about here is that the ValueError gets raised, we really don't care what the error message is.  (This is not always true, but in this case I think it is.)

In fact, in this particular case we don't even care whether we get a ValueError or a TypeError, so I'd make the test accept both as passing.  That way we're doing the same check for both str and bytes: all we are trying to prove with these tests is that we get an error rather than an empty list.